tests/acceptance/standard/test_hooks_processor.py

```diff
@@ -0,0 +1,102 @@
+import logging
+from typing import TYPE_CHECKING
+
+from gitlab.v4.objects import ProjectHook
+
+from tests.acceptance import run_gitlabform, get_random_name
+
+
+class TestHooksProcessor:
+    def project_hook(self, project):
+        name = get_random_name("hook")
+        hook = project.hooks.create({"url": f"http://hooks/{name}.org"})
+
+        return hook
+
+    def test_project_hooks_create_delete(self, gl, project):
```

Comment:
I just realized that you're using the `project` fixture. If you check out that fixture, you'll see that it's scoped to the entire test class. That means, you can actually split out your tests into separate methods like this:

```python
def test_project_hooks_create(self, gl, project):
    ...

def test_project_hooks_update(self, gl, project):
    ...

def test_project_hooks_delete(self, gl, project):
    ....
```

This way, whatever hooks are "created" in the first test case, will still be available to the second test case. After the necessary changes made in the second test case, remaining hooks (if any) will be available for the third test case.

So, basically you don't need to do the "original hooks" setup in each test case/methods. 

What you are currently doing would be suitable if you were using `project_for_function` fixture instead. That fixture is scoped to the functions. So, after each test case, the project is deleted. That means in each function you'd create some hooks first and then do what you wanted to do for the tests.

I think using the `project` fixture is fine. If you agree, let's refactor/restructure the test functions? I think it should actually simplify things a bit more.